sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp

```diff
@@ -0,0 +1,772 @@
+
+    /**
+     * @brief An AES cryptographic algorithm backed by a Hardware Security Module (HSM).
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateKeyType OctHsm;
+  };
+
+  /**
+   * @brief Elliptic Curve Cryptography (ECC) curve names.
+   *
+   */
+  class CertificateKeyCurveName final {
+  private:
+    std::string m_value;
+
+  public:
+    /**
+     * @brief Construct a new Key Curve Name object.
+     *
+     * @param value The string value of the instance.
+     */
+    explicit CertificateKeyCurveName(std::string value)
+    {
+      if (value.empty())
+      {
+        throw std::invalid_argument("The value for the curve name can not be empty");
+      }
+      m_value = std::move(value);
+    }
+
+    /**
+     * @brief Construct a default key curve.
+     *
+     */
+    CertificateKeyCurveName() = default;
+
+    /**
+     * @brief Enables using the equal operator for key curve.
+     *
+     * @param other A key curve to be compared.
+     */
+    bool operator==(const CertificateKeyCurveName& other) const noexcept
+    {
+      return m_value == other.m_value;
+    }
+
+    /**
+     * @brief Get the string value of the key curve.
+     *
+     */
+    std::string const& ToString() const { return m_value; }
+
+    /**
+     * @brief Get the NIST P-256 elliptic curve, AKA SECG curve SECP256R1.
+     *
+     * @remark For more information, see
+     *  <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateKeyCurveName P256;
+
+    /**
+     * @brief Get the SECG SECP256K1 elliptic curve.
+     *
+     * @remark For more information, see
+     * <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateKeyCurveName P256K;
+
+    /**
+     * @brief Get the NIST P-384 elliptic curve, AKA SECG curve SECP384R1.
+     *
+     * @remark For more information, see
+     * <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateKeyCurveName P384;
+
+    /**
+     * @brief Get the NIST P-521 elliptic curve, AKA SECG curve SECP521R1.
+     *
+     * @remark For more information, see
+     * <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateKeyCurveName P521;
+  };
+
+  /**
+   * @brief A collection of subject alternative names (SANs) for a X.509 certificate. SANs can be
+   * DNS entries, emails, or unique principal names.
+   *
+   */
+  struct SubjectAlternativeNameList final
```

Comment:
Nit: Other languages just use `SubjectAlternativeNames`. Is the `List` suffix here idiomatic?

Reply:
C++ can't have a class name and a field named the same.
We are using `SubjectAlternativeNames` as the field name inside the Policy.  So the Class needs to have a different name.  I just add the `List` because it contains lists inside it.  We just need a different name for the class. It can be whatever.

Reply:
@vhvb1989, you can get away with it:
```
struct SubjectAlternativeNames { int x; };

struct CertificatePolicy
{
    struct SubjectAlternativeNames SubjectAlternativeNames;
};
```